util/src/test/java/io/grpc/util/OutlierDetectionLoadBalancerTest.java

```diff
@@ -1196,6 +1285,9 @@ private void generateLoad(Map<Subchannel, Status> statusMap,
           .newClientStreamTracer(null, null);
 
       Subchannel subchannel = ((OutlierDetectionSubchannel) pickResult.getSubchannel()).delegate();
```

Comment:
We really don't want to expose HealthProducerSubchannel. This would work better:

```diff
--- a/util/src/test/java/io/grpc/util/OutlierDetectionLoadBalancerTest.java
+++ b/util/src/test/java/io/grpc/util/OutlierDetectionLoadBalancerTest.java
@@ -167,6 +167,7 @@ public class OutlierDetectionLoadBalancerTest {
       EquivalentAddressGroup eag = new EquivalentAddressGroup(addr);
       servers.add(eag);
       Subchannel sc = mock(Subchannel.class);
+      when(sc.getInternalSubchannel()).thenReturn(sc);
       subchannels.put(Arrays.asList(eag), sc);
     }
 
@@ -388,7 +389,7 @@ public class OutlierDetectionLoadBalancerTest {
     // Make sure that we can pick the single READY subchannel.
     SubchannelPicker picker = pickerCaptor.getAllValues().get(2);
     PickResult pickResult = picker.pickSubchannel(mock(PickSubchannelArgs.class));
-    assertThat(((OutlierDetectionSubchannel) pickResult.getSubchannel()).delegate()).isEqualTo(
+    assertThat(pickResult.getSubchannel().getInternalSubchannel()).isEqualTo(
         readySubchannel);
   }
 
@@ -1195,7 +1196,7 @@ public class OutlierDetectionLoadBalancerTest {
       ClientStreamTracer clientStreamTracer = pickResult.getStreamTracerFactory()
           .newClientStreamTracer(null, null);
 
-      Subchannel subchannel = ((OutlierDetectionSubchannel) pickResult.getSubchannel()).delegate();
+      Subchannel subchannel = (Subchannel) pickResult.getSubchannel().getInternalSubchannel();
 
       int maxCalls =
           maxCallsMap != null && maxCallsMap.containsKey(subchannel)
```

Long-term, we want the pickers to unwrap subchannels as it propagates pick()s. (At that point, we'd delete getInternalSubchannel().) But that can be later.